marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/WriteDataTest.java

```diff
@@ -68,45 +52,68 @@ public void testBulkIngestWithoutUriPrefix() throws IOException {
         assertFalse(uriQueryResult.hasNext());
     }
 
-    /**
-     * Spark will do all of this in the real world - i.e. a user will specify the entry class and the set of options.
-     * But in a test, we need to do that ourselves. So we create the DataSource class, build up the params, and then
-     * call the factory/writer methods ourselves.
-     *
-     * @param batchSize
-     * @param uriPrefix
-     * @return
-     */
-    private DataWriter<InternalRow> buildDataWriter(String batchSize, String uriPrefix) {
-        HubDataSource dataSource = new HubDataSource();
-        final String writeUUID = "doesntMatter";
-        final SaveMode saveModeDoesntMatter = SaveMode.Overwrite;
-
-        // Get the set of DHF properties used to connect to ML as a map, and then add connector-specific params
-        Map<String, String> params = getHubPropertiesAsMap();
-        params.put("batchsize", batchSize);
-
-        if(uriPrefix!=null && uriPrefix.length()!=0) {
-            params.put("uriprefix", uriPrefix);
-        }
+    @Test
+    public void ingestWithoutCustomApiWithCustomWorkunit(){
+        ObjectNode customWorkUnit = objectMapper.createObjectNode();
+        customWorkUnit.put("userDefinedValue", 0);
+
+        RuntimeException ex = assertThrows(RuntimeException.class,
+            () -> buildDataWriter(new Options(getHubPropertiesAsMap()).withIngestWorkUnit(customWorkUnit)),
+            "Expected an error because a custom work unit was provided without a custom API path"
+        );
+        assertEquals("Cannot set workUnit or endpointState in ingestionendpointparams unless apiPath is defined as well.", ex.getMessage());
+    }
+
+    @Test
+    public void ingestWithIncorrectApi(){
+        RuntimeException ex = assertThrows(RuntimeException.class,
+            () -> buildDataWriter(new Options(getHubPropertiesAsMap()).withIngestApiPath("/incorrect.api")),
+            "Expected an error because a custom work unit was provided without a custom API path"
+        );
+        System.out.println(ex.getMessage());
+        assertTrue( ex.getMessage().contains("Could not read non-existent document."));
+    }
+
+    @Test
+    public void ingestWithCustomApiWithCustomWorkunit() throws IOException {
```

Comment:
I think you can drop this test because WriteDataViaCustomEndpointTest already covers this with an actual custom endpoint, as opposed to using the default endpoint.